psydac/ddm/cart.py

```diff
@@ -505,8 +505,8 @@ def __init__( self, domain_decomposition, npts, global_starts, global_ends, pads
         # TODO: check that arguments are identical across all processes
         assert len( npts ) == len( global_starts ) == len( global_ends ) == len( pads ) == len(shifts)
         assert all( n >=1 for n in npts   )
-        assert all( all(n >=0) for n in global_starts )
-        assert all( all(n >=0) for n in global_ends )
+        assert all( all([ni >= 0 for ni in n]) for n in global_starts )
+        assert all( all([ni >= 0 for ni in n]) for n in global_ends )
```

Comment:
Can we not write this?
```python
        assert min(min(gs) for gs in global_starts) >= 0
        assert min(min(ge) for ge in global_ends  ) >= 0
```

Reply:
yes, sure

Reply:
done in [cacde6d](https://github.com/pyccel/psydac/pull/232/commits/cacde6d83c3713e13ab90ffe4be8cd19e19f6d37)